python/ray/ml/utils/remote_storage.py

```diff
@@ -3,14 +3,27 @@
 
 try:
     import fsspec
+
 except ImportError:
     fsspec = None
 
 try:
     import pyarrow
     import pyarrow.fs
+
+    class _CustomFSSpecHandler(pyarrow.fs.FSSpecHandler):
+        """Custom FSSpecHandler that avoids bugs in some fsspec implementations."""
```

Comment:
nit: Adding on to the TODO, I think it would be worthwhile to be explicit that this addresses a particular bug in `gcsfs` (and also rename this to `_CustomGCSFSSpecHandler`) since this isn't generally applicable to others.